pkg/apis/kubeone/types.go

```diff
@@ -923,6 +926,95 @@ type StaticAuditLogConfig struct {
 	LogMaxSize int `json:"logMaxSize,omitempty"`
 }
 
+type WebhookAuditLog struct {
+	// Enable
+	// Default value is false.
+	Enable bool `json:"enable,omitempty"`
+
+	// Config
+	Config WebhookAuditLogConfig `json:"config"`
+}
+
+type WebhookMode string
+
+const (
+	Batch          WebhookMode = "batch"
+	Blocking       WebhookMode = "blocking"
+	BlockingStrict WebhookMode = "blocking-strict"
+)
+
+type WebhookAuditLogConfig struct {
+	// PolicyFilePath is a path on local file system to the audit policy manifest
+	// which defines what events should be recorded and what data they should include.
+	// PolicyFilePath is a required field.
+	// More info: https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy
+	PolicyFilePath string `json:"policyFilePath"`
+
+	// ConfigFilePath is a path on local file system to a kubeconfig formatted file that
+	// defines how kube-apiserver can connect to the audit webhook.
+	// ConfigFilePath is a required field.
+	// More info: https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/#webhook-backend
+	ConfigFilePath string `json:"configFilePath"`
+
+	// InitialBackOff defines the amount of time to wait before retrying the first failed request.
+	// Defaults to 10s.
+	InitialBackOff metav1.Duration `json:"initialBackOff,omitempty"`
```

Comment:
@kron4eg For a follow up: I'm not the biggest fan of zero value resulting in a default value, e.g. if you don't provide anything, it's defaulted to 10s. Also, we don't default this in code, we just don't provide the flag value. This is prone to some problems like the API server changing default values, then our documentation is not going to match the reality.

We should either:
- Default to proper values in our defaulting code
- Use pointers, if nil, don't set the flag, otherwise set it (I like this one)